lib/handler/traffic.c

```diff
@@ -0,0 +1,144 @@
+#include <stdlib.h>
+#include "h2o.h"
+
+#ifndef HUNDRED_MS
+#define HUNDRED_MS 100
+#endif
+
+#ifndef ONE_SECOND
+#define ONE_SECOND 1000
+#endif
+
+typedef H2O_VECTOR(h2o_iovec_t) iovec_vector_t;
+
+typedef struct st_traffic_shaper_t {
+    h2o_ostream_t super;
+    h2o_timeout_entry_t timeout_entry;
+    size_t tokens;
+    size_t token_inc;
+    h2o_context_t *ctx;
+    h2o_req_t *req;
+    struct {
+        iovec_vector_t bufs;
+        size_t inbufcnt;
```

Comment:
You can store the number of vectors being sent in `bufs.size` instead of `inbufcnt`, and refer to `bufs.capacity` to determine if the buffer should be expanded.